ludwig/data/preprocessing.py

```diff
@@ -316,17 +317,22 @@ def preprocess_for_training(
         preprocessing_params=default_preprocessing_parameters,
         random_seed=default_random_seed
 ):
-    # Check if hdf5 and json already exist
+    data_sources = [data_df, data_train_df, data_csv, data_train_csv,
+                    data_hdf5, data_train_hdf5]
+    data_sources_none = [x is None for x in data_sources]
+    if all(data_sources_none):
```

Comment:
This becomes `any`

Reply:
I'll have to change it to 'if not any' right?

Reply:
yes

Reply:
done